### PR TITLE
chore: RLN contract unit test expansion

### PR DESCRIPTION
**Pull request by romanzac**
## Description

Add edge cases to improve unit tests coverage. 

## Issues reported

- https://github.com/waku-org/waku-rlnv2-contract/issues/32

## Tests added

- test__ErasingNonExistentMembership
- test__GracePeriodExtensionEdgeCases
- test__MaxTotalRateLimitEdgeCases
- test__MerkleTreeUpdateAfterErasureAndReuse
- test__ZeroGracePeriodDuration
- test__FullCleanUpErasure
- test__TokenTransferFailures
- test__ReinitializationProtection
- test__PriceCalculatorReconfiguration
- test__ZeroPriceEdgeCase

## Tests not added

- test__ReentrancyProtectionRegister
- test__ReentrancyProtectionWithdraw

## Checklist

Ensure you completed **all of the steps** below before submitting your pull request:

- [x] Added natspec comments?
- [x] Ran `pnpm adorno`?

### CODE REVIEW

File: test/WakuRlnV2.t.sol [hunk 1]
```diff
@@ -36,6 +117,17 @@ contract WakuRlnV2Test is Test {
...
+        // Log owner for debugging
+        console.log("WakuRlnV2 owner: ", w.owner());
+
+        // Transfer ownership to address(this)
+        vm.prank(w.owner());
+        try w.transferOwnership(address(this)) {
+            console.log("Ownership transferred to: ", w.owner());
+        } catch {
+            console.log("Failed to transfer ownership");
+        }
```
Comment (0x-r4bbit) [inline test/WakuRlnV2.t.sol:129]: Any reason this is done?

File: test/WakuRlnV2.t.sol [hunk 2]
```diff
@@ -789,4 +881,419 @@ contract WakuRlnV2Test is Test {
...
+        uint32 rateLimit = w.minMembershipRateLimit();
+        (, uint256 price) = w.priceCalculator().calculate(rateLimit);
+
+        // Approve tokens
+        maliciousToken.approve(address(w), price);
+
+        // Expect transfer failure
+        vm.expectRevert("Malicious transfer failure");
+        w.register(1, rateLimit, noIdCommitmentsToErase);
+    }
```
Comment (0x-r4bbit) [inline test/WakuRlnV2.t.sol:1183]: While this test checks if malicious token does indeed revert, it's not actually testing any functionality on the rln contract itself. 

So I'm not sure it adds any value.

File: test/WakuRlnV2.t.sol [hunk 3]
```diff
@@ -789,4 +881,419 @@ contract WakuRlnV2Test is Test {
...
+        // 2) Prepare args BEFORE expectRevert (to avoid consuming it with view calls)
+        address calc = before_.priceCalculator;
+        uint32 maxTotal = before_.maxTotalRateLimit;
+        uint32 minRate = before_.minMembershipRateLimit;
+        uint32 maxRate = before_.maxMembershipRateLimit;
+        uint32 activeDur = 15;
+        uint32 graceDur = 5;
+
+        // 3) Second initialization must revert (use a loose matcher for OZ v4/v5 compatibility)
+        vm.expectRevert();
```
Comment (0x-r4bbit) [inline test/WakuRlnV2.t.sol:1228]: If possible, expect exact error type/message

**Review by 0x-r4bbit — COMMENTED** [review]
Great battery of tests!
Good work @romanzac

File: test/WakuRlnV2.t.sol (hunk 3 above)
Comment (romanzac) [reply test/WakuRlnV2.t.sol]: Fixed at https://github.com/waku-org/waku-rlnv2-contract/pull/31/commits/8e9814c8912e9c84dd01f78c9848e80e9df6da7a

File: test/WakuRlnV2.t.sol (hunk 2 above)
Comment (romanzac) [reply test/WakuRlnV2.t.sol:1149]: It is from similar category like were the reentrancy tests. If during deployment or reconfiguration owner enters by accident wrong address with malicious token contract, it could lead to lost funds. There is a small chance, when the deployment scripts sit on public repo, that repo will get hacked and deployment will happen to malicious contract.

Comment (romanzac) [reply test/WakuRlnV2.t.sol:1149]: I've cleanup up the MaliciousToken from reentrancy capabilities. https://github.com/waku-org/waku-rlnv2-contract/pull/31/commits/fb9f55b6cd63de22c1702ffba067019ed49f911a
